src/Symfony/Component/Routing/Route.php

```diff
@@ -86,6 +86,13 @@ class Route implements \Serializable
      */
     public function __construct($path, array $defaults = array(), array $requirements = array(), array $options = array(), $host = '', $schemes = array(), $methods = array(), $condition = '')
     {
+        if (isset($requirements['_method']) || isset($requirements['_scheme'])) {
```

Comment:
this should be done in ``setRequirements`` actually (given that the constructor calls it)